index.ts

```diff
@@ -140,6 +140,53 @@ const mem = <
 
 export = mem;
 
+/**
+Returns a decorator which memoizes the function provided.
+
+@example
+```
+import mem = require('mem');
+
+class Example {
+	constructor() {
+		this.i = 0;
+	}
+
+	@mem.decorator()
+	counter() {
+		return ++this.i;
+	}
+}
+
+class ExampleWithOptions {
+	constructor() {
+		this.i = 0;
+	}
+
+	@mem.decorator({maxAge: 1000})
+	counter() {
+		return ++this.i;
+	}
+}
+```
+*/
+mem.decorator = <
+	FunctionToMemoize extends AnyFunction,
+	CacheKeyType
+>(
+	options: Options<FunctionToMemoize, CacheKeyType> = {}
+) => (
+	target: any,
```

Comment:
I don't think there's a way to avoid this being `any`.

Reply:
`target` refers to the object that the function lives in. `const input: FunctionToMemoize = target[propertyKey]`.

Reply:
Shouldn't it at least be `object` then to ensure `undefined` and `null` are not the target? Also, I generally prefer `unknown` over `any` whenever possible.

Reply:
`object` isn't allowed by `xo` which wants it to be `Record<string, unknown>` which doesn't work either because a class is not an plain object.

Reply:
Not using `object` is just a general rule. You shouldn't use `any` either, ideally. But sometimes we need exceptions. Just use an `eslint-ignore` comment.

Reply:
As soon as I change `any` to `object`, the compiler throws because [`{}` can't be indexed by a `string`](https://stackoverflow.com/questions/48382306/typescript-how-skip-define-type-on-multiple-generics-class).

If I try to use generics, the compiler is instead unable to resolve a target properly so it always falls back to `unknown`.